ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewModel.kt

```diff
@@ -6,19 +6,36 @@ import androidx.lifecycle.ViewModel
 
 class PreviewImageViewModel : ViewModel() {
     private lateinit var lowResImageUrl: String
+    private lateinit var highResImageUrl: String
 
-    private val _loadImageFromUrl = MutableLiveData<String>()
-    val loadImageFromUrl: LiveData<String> = _loadImageFromUrl
+    private val _loadImageFromData = MutableLiveData<ImageData>()
+    val loadImageFromData: LiveData<ImageData> = _loadImageFromData
 
     private var isStarted = false
 
-    fun start(loResImageUrl: String) {
+    fun start(loResImageUrl: String, hiResImageUrl: String) {
         if (isStarted) {
             return
         }
         isStarted = true
 
         lowResImageUrl = loResImageUrl
-        _loadImageFromUrl.value = lowResImageUrl
+        highResImageUrl = hiResImageUrl
+
+        _loadImageFromData.value = ImageData(
+            uiState = ImageUiState.IMAGE_LOAD_IN_PROGRESS,
+            lowResImageUrl = lowResImageUrl,
+            highResImageUrl = highResImageUrl
+        )
+    }
+
+    data class ImageData(val uiState: ImageUiState, val lowResImageUrl: String, val highResImageUrl: String)
```

Comment:
Thinking about it a bit more, I'd even consider changing direction of the dependency from ImageUiState ->ImageData, to ImageData->ImageUiState. I basically think, only `IMAGE_LOAD_IN_PROGRESS` needs to have access to the ImageData, right? (we'd need to replace enum with a sealed class).

Reply:
Replaced here 2a39251, 30f2166